cf-revision-manager.php

```diff
@@ -2,200 +2,450 @@
+
+/**
+ * Revision Manger admin form callback function.
+ *
+ * @since  2.0.0
+ *
+ * @return void
+ */
+function cfrm_admin_form() {
+	//include_once plugin_dir_path( __FILE__ ) . 'includes/admin-form.php';
+	wp_enqueue_style( 'cfrm-style' );
+	wp_enqueue_script( 'cfrm-script' );
+	$required_keys = cfrm_set_required_keys();
+	$keys = array_diff( cfrm_get_distinct_meta_keys(), cfrm_excluded_keys(), $required_keys );
+?>
+<div class="wrap">
+	<h2><?php _e( 'CF Revision Manager', 'cfrm' ); ?></h2>
+<?php
+		if ( ! count( $keys ) ) {
+			echo '<p>' . __( 'No custom fields found.', 'cfrm' ) . '</p>';
+		} else {
+			echo '<form id="cfr_revision_manager_form" name="cfr_revision_manager_form" action="' . admin_url( 'options-general.php' ) . '" method="post">
+				<p>' . __( 'A plugin or theme has specified that the following custom fields need to included in revisions.', 'cfrm' ) . '</p>';
+			if ( count( $required_keys ) ) {
+				echo '<div>
+				<ul id="cfr_revision_manager_keys_required">';
+				foreach ( $required_keys as $key ) {
+					$checked = $key;
+					$disabled = $key;
+					$id = 'cf_revision_manager_key_' . esc_attr( $key );
+					echo '<li>
+						<input type="checkbox" name="revision_manager_keys[]" id="' . $id . '" value="' . esc_attr( $key ) . '" ' . checked( $key, $checked, false ) . ' ' . disabled( $key, $disabled, false ) . ' />
+						<label for="' . $id . '">' . esc_html( $key ) . '</label>
+					</li>';
 				}
+				echo '</ul>
+				</div>';
 			}
-		}
-	
-		public function register($postmeta_key, $display_func = '') {
-			if (!in_array($postmeta_key, $this->postmeta_keys, true)) {
-				$this->postmeta_keys[] = compact('postmeta_key', 'display_func');
+			echo '<p class="clearfix">' . __( 'Below is a list of selectable custom fields for this site. Choose the ones you would like to have included in your revisions.', 'cfrm' ).'</p>
+				<div>
+				<ul id="cfr_revision_manager_keys">';
+			foreach ( $keys as $key ) {
+				$checked = ( in_array( $key, cfrm_get_selected_keys() ) ? $key : '' );
+				$disabled = '';
+				$id = 'cf_revision_manager_key_' . esc_attr( $key );
+				echo '<li>
+					<input type="checkbox" name="revision_manager_keys[]" id="' . $id . '" value="' . esc_attr( $key ) . '" ' . checked( $key, $checked, false ) . ' ' . disabled( $key, $disabled, false ) . ' />
+					<label for="' . $id . '">' . esc_html( $key ) . '</label>
+				</li>';
 			}
-			return true;
-		}
-	
-		/**
-		 * This is a paranoid check. There will be no object to register the 
-		 * actions and filters if nobody adds any postmeta to be handled
-		 *
-		 * @return bool
-		 */
-		public function have_keys() {
-			return (bool) count($this->postmeta_keys);
+			echo '</ul>
+				</div>
+				<p class="submit">
+				<input type="submit" name="submit_button" class="button-primary" value="' . __( 'Save', 'cfrm' ) . '" />
+				</p>
+				<input type="hidden" name="cf_action" value="cfr_save_keys" class="hidden" style="display: none;" />
```

Comment:
The `class="hidden" style="display:none;" are totally unnecessary for a hidden input.

In fact, this entire hidden input should be unnecessary if we're properly looking for and validating the nonce as part of our save routine.